core/tests/ts-integration/tests/api/web3.test.ts

```diff
@@ -355,7 +355,6 @@ describe('web3 API compatibility tests', () => {
         // ...though the gap should not be *too* big.
         expect(newBlock).toBeLessThan(currentBlock + 100);
         await tx.wait(); // To not leave a hanging promise.
-        wsProvider.removeAllListeners();
```

Comment:
This call seems useless if not awaited (the unsubscribe RPC requests scheduled internally likely won't be sent since the WebSocket is closed immediately). But if it is awaited, then `destroy()` below will fail because unsubscribe RPC requests get rejected. I haven't found a good way to wait until these requests are sent (something like `await wsProvider.sendPendingRequests()`), so the only option seems to have a hard-coded timeout, which doesn't look good.